### PR TITLE
feat(blade): Single radio - PR1

### PR DESCRIPTION
**Pull request by anuraghazra**
Not ready for review. 

Added single radio component.

### CODE REVIEW

File: packages/blade/src/components/Radio/Radio.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    isRequired,
+    name,
+    value,
+    onChange,
+  });
+  const _hasError = validationState === 'error';
+  const showHelpText = !_hasError && helpText;
+
+  return (
+    <SelectorLabel inputProps={state.isReactNative ? inputProps : {}}>
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/Radio.tsx:102]: Instead of returning `isReactNative` from the `useRadio` hook just directly use `getPlatformType()`. `isReactNative` is not logically a state for a Radio, right?
```suggestion
    <SelectorLabel inputProps={getPlatformType() === 'react-native' ? inputProps : {}}>
```

File: packages/blade/src/components/Radio/Radio.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      <CheckboxInput
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/Radio.tsx:103]: Why are we using `CheckboxInput`. If this is a common component then shouldn't we move this to Forms & call it `Input` and use it in both checkbox & radio?

File: packages/blade/src/components/Radio/Radio.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        isChecked={state.isChecked}
+        isDisabled={isDisabled}
+        isNegative={_hasError}
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/Radio.tsx:106]: Why not just have the prop as `hasError`? `isNegative` is another verbatim that we'll have to remember. For input whenever it's negative it has error so might as well just call is `hasError`?
```suggestion
        hasError={_hasError}
```

File: packages/blade/src/components/Radio/Radio.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        inputProps={inputProps}
+      />
+      <RadioIcon isChecked={state.isChecked} isDisabled={isDisabled} isNegative={_hasError} />
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/Radio.tsx:109]: Same here. Should we call this `hasError` too? There are 3 variables controlling an error state
- validationState
- hasError
- isNegative

I understand why we don't want to use validation state since it's a string check and boolean is more convenient so at least lets stick to one boolean only? `validationState='error'` & `hasError` seem related. isNegative seems a bit out of the odd. 
```suggestion
      <RadioIcon isChecked={state.isChecked} isDisabled={isDisabled} hasError={_hasError} />
```

File: packages/blade/src/components/Radio/Radio.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      <Box>
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/Radio.tsx:110]: Do we need a `Box` here or can we just use `React.Fragment` here?

File: packages/blade/src/components/Radio/RadioIcon.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import React from 'react';
+import { useTheme } from '~components/BladeProvider';
+import Svg from '~components/Icons/_Svg';
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/RadioIcon.tsx:5]: nit: We should do named exports here too since we've decided to do it everywhere 😅 

We can do this later as well. Not in this PR's scope

File: packages/blade/src/components/Radio/RadioIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+import { getIn } from '~utils';
+import { CheckboxIconWrapper } from '~components/Checkbox/CheckboxIcon/CheckboxIconWrapper';
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/RadioIcon.tsx:7]: Lets we extract this to a common place now? `InputIconWrapper`?

File: packages/blade/src/components/Radio/useRadio.ts [hunk 8]
```diff
@@ -0,0 +1,106 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import React from 'react';
+import type { GestureResponderEvent } from 'react-native';
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/useRadio.ts:4]: Importing from `react-native` in a file that can be used on web is dangerous. This will cause failures on projects that are react only and wouldn't be able to find `react-native`. Lets not do this?

File: packages/blade/src/components/Radio/useRadio.ts [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+import { useControllableState } from '../../hooks/useControllable';
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/useRadio.ts:5]: use import aliases?

File: packages/blade/src/components/Radio/useRadio.ts [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+  onChange,
+  name,
+  value,
+}: UseRadioProps) => {
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const isReactNative = getPlatformType() === 'react-native';
+  if (isChecked && defaultChecked) {
+    throw new Error(
+      `[Blade useRadio] Do not provide both 'isChecked' and 'defaultChecked' to useRadio. Consider if you want this component to be controlled or uncontrolled.`,
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/useRadio.ts:30]: If I as a consumer pass `isChecked` and `defaultChecked` will I get the same error? Seems misleading for a consumer since it talks about `useRadio`. If this is consumer facing then we should rephrase this to make it more generic for Radio

File: packages/blade/src/components/Radio/useRadio.ts [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+    );
+  }
+
+  const [radioState, setRadioStateChange] = useControllableState({
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/useRadio.ts:34]: What does `setRadioStateChange` mean? The name seems too confusing. Can we have a more meaningful name here or add comments to indicate what this does?

File: packages/blade/src/components/Radio/useRadio.ts [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+    value: isChecked,
+    defaultValue: defaultChecked ?? false,
+  });
+
+  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement> | GestureResponderEvent) => {
+    if (isDisabled) {
+      event.stopPropagation();
+      event.preventDefault();
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/useRadio.ts:42]: These might not be available on react native. Can you check and ensure this doesn't break on react native?

File: packages/blade/src/components/Radio/useRadio.ts [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+        isChecked: !checked,
+        event: event as React.ChangeEvent,
+        value,
+      });
+      return !checked;
+    });
+  };
+
+  const state = {
+    isReactNative,
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/useRadio.ts:57]: Don't think this belongs here. Let the consumer component call `getPlatformType` if it needs to figure out the platform

File: packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts [hunk 14]
```diff
@@ -38,10 +38,12 @@ const variants = {
 
 export type CheckboxRectProps = Omit<CheckboxIconProps, 'state'> & {
   isChecked: boolean;
+  isFullRounded?: boolean;
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts:41]: Just `isRounded`?
```suggestion
  isRounded?: boolean;
```

File: packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts [hunk 15]
```diff
@@ -38,10 +38,12 @@ const variants = {
 
 export type CheckboxRectProps = Omit<CheckboxIconProps, 'state'> & {
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts:39]: We should extract this component out into a common place and not import from Checkbox for Radio

**Review by chaitanyadeorukhkar — CHANGES_REQUESTED** [review]

File: packages/blade/src/components/Radio/Radio.tsx [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+  defaultChecked,
+  isChecked,
+  isDisabled,
+  isRequired,
+  name,
+  value,
+  onChange,
+  children,
+  helpText,
+  validationState,
```
[the remark this answers is not in the document]
Comment (anuraghazra) [reply packages/blade/src/components/Radio/Radio.tsx:87]: Will do, tho can't put on defaultChecked, isChecked since absence of them (undefined) will determine the behaviour too.

File: packages/blade/src/components/Radio/Radio.tsx (hunk 1 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/Radio.tsx:102]: Yeah right I guess we should do that.

File: packages/blade/src/components/Radio/Radio.tsx (hunk 2 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/Radio.tsx:103]: Fixed in next PRs which has FormLabel stuff

File: packages/blade/src/components/Radio/Radio.tsx (hunk 3 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/Radio.tsx:106]: 🤔 Did it because in here isNegative is a intent not a state. It just changes the visual appearance.

File: packages/blade/src/components/Radio/Radio.tsx (hunk 4 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/Radio.tsx:109]: Okay then, that makes sense. I'll made this change on #607 since these components are mostly either moved or renamed. I'll add a TODO.

File: packages/blade/src/components/Radio/Radio.tsx (hunk 5 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/Radio.tsx:110]: We do need Box, since we need to control the flex layout.

File: packages/blade/src/components/Radio/RadioIcon.tsx (hunk 6 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/RadioIcon.tsx:5]: Yeah let's do this on a separate PR.

File: packages/blade/src/components/Radio/RadioIcon.tsx (hunk 7 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/RadioIcon.tsx:7]: Yep already done in #607

File: packages/blade/src/components/Radio/useRadio.ts (hunk 8 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/useRadio.ts:4]: It's a `type` import.

File: packages/blade/src/components/Radio/useRadio.ts (hunk 9 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/useRadio.ts:5]: done

File: packages/blade/src/components/Radio/useRadio.ts (hunk 10 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/useRadio.ts:30]: Sure.

File: packages/blade/src/components/Radio/useRadio.ts (hunk 11 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/useRadio.ts:34]: Yeah right 😆

File: packages/blade/src/components/Radio/useRadio.ts (hunk 12 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/useRadio.ts:42]: Nope doesn't seem to break, This code is also present on Checkbox. I think React's synthetic event system mocks these on native.

File: packages/blade/src/components/Radio/useRadio.ts (hunk 13 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/useRadio.ts:57]: Removed.

File: packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts (hunk 15 above)
Comment (anuraghazra) [reply packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts:39]: Done in PR2

File: packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts [hunk 17]
```diff
@@ -38,10 +38,12 @@ const variants = {
...
   isChecked: boolean;
+  isFullRounded?: boolean;
 };
 
 const getCheckboxIconWrapperStyles = ({
   theme,
+  isFullRounded,
```
Comment (divyanshu013) [inline packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts:46]: Think you need to update the instances of `isFullRounded`

File: packages/blade/src/components/Icons/_Svg/Circle/Circle.d.ts [hunk 18]
```diff
@@ -0,0 +1,11 @@
+export type CircleProps = {
```
Comment (divyanshu013) [inline packages/blade/src/components/Icons/_Svg/Circle/Circle.d.ts:1]: Would all be optional?

File: packages/blade/src/components/Radio/Radio.stories.tsx [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+import type { ComponentStory, Meta } from '@storybook/react';
+import { Button, Text } from '..';
+import type { RadioProps } from './Radio';
+import { Radio as RadioComponent } from './Radio';
+import Box from '~components/Box';
+
+export default {
+  title: 'Components/Radio/Radio',
+  component: RadioComponent,
+  args: {
```
Comment (divyanshu013) [inline packages/blade/src/components/Radio/Radio.stories.tsx:12]: I thought we don't need to specify `undefined` props below 🤔

File: packages/blade/src/components/Radio/Radio.tsx [hunk 20]
```diff
@@ -0,0 +1,120 @@
+/* eslint-disable @typescript-eslint/no-shadow */
```
Comment (divyanshu013) [inline packages/blade/src/components/Radio/Radio.tsx:1]: Instead shouldn't we use variables with different names in local scope?

File: packages/blade/src/components/Radio/Radio.tsx [hunk 21]
```diff
@@ -0,0 +1,120 @@
...
+    isChecked,
+    isDisabled,
+    isRequired,
+    name,
+    value,
+    onChange,
+  });
+  const _hasError = validationState === 'error';
+  const showHelpText = !_hasError && helpText;
+  const isReactNative = getPlatformType() === 'react-native';
```
Comment (divyanshu013) [inline packages/blade/src/components/Radio/Radio.tsx:101]: Can we move platform specific logic to respective files? .web, .native i.e.

File: packages/blade/src/components/Radio/useRadio.ts (hunk 8 above)
Comment (divyanshu013) [reply packages/blade/src/components/Radio/useRadio.ts:4]: But it's coming from `react-native` package which won't exist on web only projects

Comment (divyanshu013) [reply packages/blade/src/components/Radio/useRadio.ts:4]: I'm in favor of having two different modules .web, .native for this. Thoughts?

File: packages/blade/src/components/Radio/useRadio.ts [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+  isDisabled,
+  isRequired,
+  hasError,
+  onChange,
+  name,
+  value,
+}: UseRadioProps) => {
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const isReactNative = getPlatformType() === 'react-native';
```
Comment (divyanshu013) [inline packages/blade/src/components/Radio/useRadio.ts:27]: Can we move this logic to appropriate .web / .native?

**Review by divyanshu013 — COMMENTED** [review]
Some queries. Also it seems the checkbox and text aren't perfectly horizontal center aligned, is it just me?

image

File: packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts (hunk 17 above)
Comment (anuraghazra) [reply packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts:46]: Why?

File: packages/blade/src/components/Icons/_Svg/Circle/Circle.d.ts (hunk 18 above)
Comment (anuraghazra) [reply packages/blade/src/components/Icons/_Svg/Circle/Circle.d.ts:1]: Probably some will be required.

File: packages/blade/src/components/Radio/Radio.stories.tsx (hunk 19 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/Radio.stories.tsx:12]: Yeah wouldn't matter if we don't, just the sorting will be random.

File: packages/blade/src/components/Radio/Radio.tsx (hunk 21 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/Radio.tsx:101]: Not really feasible. 

We are just adding a prop: 

```
inputProps={isReactNative ? inputProps : {}}
```

Duplicating and maintaining two identical component just for a simple prop can be tricky.

File: packages/blade/src/components/Radio/useRadio.ts (hunk 8 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/useRadio.ts:4]: > But it's coming from react-native package which won't exist on web only projects

type imports are stripped away and won't even be imported/bundled on the web project. 

Anyway we will have this on each and every component when we have TS 4.7 module suffixes.

File: packages/blade/src/components/Radio/useRadio.ts (hunk 22 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/useRadio.ts:27]: Again, not feasible. 
Imagine maintaining useRadio twice with identical logic. 

.web / .native files makes more sense when there's little to know overlap or lets say when we use the Styled* components.

File: packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts (hunk 17 above)
Comment (divyanshu013) [reply packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts:46]: Because we're calling it `isRounded`? [Ref](https://github.com/razorpay/blade/pull/605#discussion_r944391741)

**Review by divyanshu013 — APPROVED** [review]
Looks good to me.

File: packages/blade/src/components/Radio/useRadio.ts (hunk 8 above)
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Radio/useRadio.ts:4]: Since its only type and its going to be stripped off, I think its okay. With TS 4.7 we'll see what to do

**Review by chaitanyadeorukhkar — APPROVED** [review]

File: packages/blade/src/components/Form/Selector/SelectorGroupField.tsx [hunk 23]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import type { CSSObject } from 'styled-components';
+import { getPlatformType, makeAccessible } from '~utils';
+import Box from '~components/Box';
+
+type SelectorGroupFieldProps = {
+  children: React.ReactNode;
+  labelledBy: string;
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Form/Selector/SelectorGroupField.tsx:8]: Can we stick to american english since we're doing that everywhere? 😅 
```suggestion
  labeledBy: string;
```

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Form/Selector/SelectorGroupField.tsx:8]: Or are we doing this cuz of a11y props?

File: packages/blade/src/components/Form/Selector/SelectorInput.web.tsx [hunk 24]
```diff
@@ -50,12 +52,12 @@ const StyledInput = styled.input<HoverProps>(({ theme, isChecked, isDisabled, is
...
 }));
 
-const CheckboxInput = ({
+const SelectorInput = ({
   inputProps,
   isChecked,
   isDisabled,
   isNegative,
-}: HoverProps & { inputProps: InputProps }): React.ReactElement => {
+}: HoverProps & { inputProps: any }): React.ReactElement => {
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Form/Selector/SelectorInput.web.tsx:60]: Oh no :(

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Form/Selector/SelectorInput.web.tsx:60]: No other way around?

File: packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts [hunk 25]
```diff
@@ -0,0 +1,102 @@
...
+  const idBase = useId('radio-group');
+  const labelId = `${idBase}-label`;
+  const fallbackName = name ?? idBase;
+  const [checkedValue, setValue] = useControllableState({
+    value,
+    defaultValue,
+    onChange: (v: string) => onChange?.({ value: v, name: fallbackName }),
+  });
+
+  // FIX React native bug, where it unchecks the radio on clicking it again
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts:54]: Is this fixed?

File: packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts [hunk 26]
```diff
@@ -0,0 +1,102 @@
...
+        }
+
+        setValue(() => v);
+      },
+      removeValue(): void {
+        if (isDisabled) {
+          return;
+        }
+
+        // @ts-expect-error TODO: fix this
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts:70]: Looks like a pending todo. Can you take a look?

File: packages/blade/src/components/Radio/RadioIcon/Fade.native.tsx [hunk 27]
```diff
@@ -0,0 +1,53 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import React from 'react';
+import type { EasingFn } from 'react-native-reanimated';
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/RadioIcon/Fade.native.tsx:4]:
```suggestion
import type { EasingFunctionFactory } from 'react-native-reanimated';
```

File: packages/blade/src/components/Radio/RadioIcon/Fade.native.tsx [hunk 28]
```diff
@@ -0,0 +1,53 @@
...
+  return {
+    ...styles,
+  };
+});
+
+const Fade = ({ children, show, styles }: FadeProps) => {
+  const { theme } = useTheme();
+
+  const enterEasing = (theme.motion.easing.entrance.effective as unknown) as EasingFn;
+  const exitEasing = (theme.motion.easing.exit.effective as unknown) as EasingFn;
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/RadioIcon/Fade.native.tsx:21]:
```suggestion
  const enterEasing = (theme.motion.easing.entrance.effective as unknown) as EasingFunctionFactory;
  const exitEasing = (theme.motion.easing.exit.effective as unknown) as EasingFunctionFactory;
```

File: packages/blade/src/components/Form/Selector/SelectorGroupField.tsx (hunk 23 above)
Comment (anuraghazra) [reply packages/blade/src/components/Form/Selector/SelectorGroupField.tsx:8]: > Can we stick to american english since we're doing that everywhere? 😅

our `makeAccessible` follows the WAI-ARIA props which takes british `labelledBy`, let's keep it as is for consistency sake. 

> Or are we doing this cuz of a11y props?

Yes so that the group has a label reference of it's child text. It works similarly to `fieldset` & `legend`

```tsx
<fieldset>
  <legend>hello</legend>
</fieldset>

<SelectorGroupField labelledBy="id-1">
  <Text id="id-1">hello</Text>
</SelectorGroupField>
```

File: packages/blade/src/components/Radio/RadioIcon/Fade.web.tsx [hunk 29]
```diff
@@ -0,0 +1,74 @@
...
+    animation: ${scaleIn} ${makeMotionTime(duration)}
+      ${theme.motion.easing.entrance.effective as string};
+  `;
+
+  const exit = css`
+    animation: ${fadeOut} ${makeMotionTime(duration)}
+      ${theme.motion.easing.exit.effective as string};
+  `;
+
+  const { isMounted, isVisible } = usePresence(!!show, {
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/RadioIcon/Fade.web.tsx:58]: Can you add some comments here to help understand what we're doing with this?

File: packages/blade/src/components/Form/Selector/SelectorInput.web.tsx (hunk 24 above)
Comment (anuraghazra) [reply packages/blade/src/components/Form/Selector/SelectorInput.web.tsx:60]: Could be, but fine for now. Anyways this component is only shared in Checkbox & radio for now. which has similar APIs

File: packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts (hunk 25 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts:54]: Oh yeah I did fixed it, let me check. Maybe merge kerne ke time pe it got messed up.

File: packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts [hunk 30]
```diff
@@ -0,0 +1,76 @@
...
+  return {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    width: '16px',
+    height: '16px',
+    borderWidth: '1px',
+    borderStyle: 'solid',
+    margin: '0px',
+    marginTop: '3px',
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts:64]: Margin of 3px? Where is this?

File: packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts [hunk 31]
```diff
@@ -0,0 +1,76 @@
...
+  const backgroundColor = variants[variant].background[checked];
+  const borderColor = variants[variant].border[checked];
+
+  return {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    width: '16px',
+    height: '16px',
+    borderWidth: '1px',
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts:61]: Can we not use border tokens?

File: packages/blade/src/components/Radio/RadioIcon/Fade.native.tsx (hunk 27 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/RadioIcon/Fade.native.tsx:4]: Can't do that. Keyframe() takes EasingFn only.

File: packages/blade/src/components/Radio/RadioIcon/Fade.web.tsx (hunk 29 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/RadioIcon/Fade.web.tsx:58]: Will do.

File: packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts [hunk 32]
```diff
@@ -0,0 +1,76 @@
...
+    borderStyle: 'solid',
+    margin: '0px',
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts:63]:
```suggestion
    margin: makeSpace(theme.spacing[0])
```

File: packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts (hunk 30 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts:64]: Not a token from figma. It's to vertical align the Icon to text. We can't use flex since It has to stay in the center of text not the whole wrapper.

File: packages/blade/src/components/Radio/useRadio.ts [hunk 33]
```diff
@@ -0,0 +1,158 @@
...
+  if (isChecked && defaultChecked) {
+    throw new Error(
+      `[Blade Radio] Do not provide both 'isChecked' and 'defaultChecked' to useRadio. Consider if you want this component to be controlled or uncontrolled.`,
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/useRadio.ts:78]: Will this error be shown to the user if they pass `isChecked` & `defaultChecked` together? If yes then we need to rephrase this. User shouldn't be worried about what `useRadio` is

```suggestion
      `[Blade: Radio] Do not provide both 'isChecked' and 'defaultChecked' to useRadio. Consider if you want this component to be controlled or uncontrolled.`,
```

File: packages/blade/src/components/Radio/Radio.stories.tsx [hunk 34]
```diff
@@ -0,0 +1,159 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import type { ComponentStory, Meta } from '@storybook/react';
+import React from 'react';
+import { Text } from '../Typography';
+import type { RadioGroupProps } from './RadioGroup/RadioGroup';
+import { RadioGroup as RadioGroupComponent } from './RadioGroup/RadioGroup';
+import { Radio as RadioComponent } from './Radio';
+
+export default {
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/Radio.stories.tsx:9]: Check the stories for `Checkbox` once. There are some things missing here like Subtitle, Usage & Example. Lets add that to make the documentation consistent

File: packages/blade/src/components/Radio/Radio.stories.tsx [hunk 35]
```diff
@@ -0,0 +1,159 @@
...
+  title: 'Components/Radio',
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/Radio.stories.tsx:10]: Lets also mention RadioGroup here so consumers know that this is a combined doc?
```suggestion
  title: 'Components/Radio & RadioGroup',
```

File: packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts (hunk 25 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts]: @chaitanyadeorukhkar Yeah I did fixed it. In [here ](https://github.com/razorpay/blade/pull/605/files#diff-82055f61296921ac0c865ece1e8c254f6ebd3bdf0e2dee1c33d2244cfdac1634R95-R96)

File: packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts (hunk 26 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts:69]: Just a TS error. setValue() doesn't accept undefined but I want it to, to be able to remove the selected radio. Should be fine.

File: packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts (hunk 25 above)
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts]: Remove the comment then?

File: packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts (hunk 30 above)
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts:64]: This could be troublesome. 3px might not stay true to align it to the center in various circumstances - platform (android, ios, web), pixel density, a11y font changes, etc. Have you verified the alignment on Android as well as iOS?

File: packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts (hunk 31 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts:61]: We can.

File: packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts (hunk 30 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts:64]: I see yeah could be,  I don't see any difference in native. Anyway are we doing pixel density calculation in any of our spacing tokens?

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts:71]: No but we shouldn't use margin to center align it. We should rely on some form of layout
